qlib/Logger.qm

```diff
@@ -343,15 +379,18 @@ public namespace Logger {
             @throw LOGGER-ERROR if defaultLevel is nothing and level not found
         */
         public static LoggerLevel getLevel(int level, *LoggerLevel default_level) {
-            if (exists LEVEL_TO_LEVEL_STR{level}) {
+            try {
+                Logger::LoggerLevel::levelIntToStr(level);
+
                 if (!exists levelMap{level}) {
-                    levelMap{level} = new LoggerLevel(level, LEVEL_TO_LEVEL_STR{level});
+                    levelMap{level} = new LoggerLevel(level);
                 }
                 return levelMap{level};
-            } else {
-                if (!exists default_level) {
-                    throw "LOGGER-ERROR", "Cannot get valid Level";
+            } catch (hash<ExceptionInfo> ex) {
```

Comment:
I would not do this catching but create rather e.g. Logger::LoggerLevel::levelIntToStrIntern() not throwing exception and Logger::LoggerLevel::levelIntToStr will call Logger::LoggerLevel::levelIntToStrIntern+exception.